### PR TITLE
Doco on how to optimize message processing

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
@DavidBoike After using @wolfbyte as my rubber duck he came up with Optimization instead of Throttling. I've reworded the pull accordingly, thoughts?

cc @Particular/tf-asyncawait

File: nservicebus/operations/tuning.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+related:
+- nservicebus/licensing/licensing-limitations
+---
+NServiceBus will by default allow the transport to optimize for maximum performance when it comes to message processing. While this is usually the preferred mode of operation there are situations where tuning needs to be applied. 
+
+
+## Tuning concurrency
+You can define a maximum concurrency setting that will make sure that no more messages than the specified value is ever being processed at the same time. Set this value to `1` to process messages sequentially. If not specified the transport will choose an optimal value.  
+
+Examples where concurrency tuning is relevant are non thread safe code that needs to run sequentially. Databases that might deadlock when getting to many concurrent requests etc.
```
Comment (SeanFeldman) [inline nservicebus/operations/tuning.md]: This is hard to read. Should it be a bullet list?
```
Examples where concurrency tuning is relevant are:

- Non thread safe code that needs to run sequentially
- Databases that might deadlock when getting to many concurrent requests
- etc
```

File: nservicebus/operations/tuning.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## Tuning concurrency
+You can define a maximum concurrency setting that will make sure that no more messages than the specified value is ever being processed at the same time. Set this value to `1` to process messages sequentially. If not specified the transport will choose an optimal value.  
+
+Examples where concurrency tuning is relevant are non thread safe code that needs to run sequentially. Databases that might deadlock when getting to many concurrent requests etc. 
+
+NOTE: NServiceBus Version 5 and below will by default limit concurrency to `1` if not configured by the user
+
+## Tuning throughput
+
+You can define a maximum value for the number of messages per second that the endpoint will process at any given time. This will help you to avoid your endpoint overloading sensitive resources that it's using like web-services, databases, other endpoints etc. A concrete example here could be an integration endpoint calling a rate limited web api.
```
Comment (SeanFeldman) [inline nservicebus/operations/tuning.md]: "avoid your endpoint overloading" --> "avoid your endpoint from overloading"
"like web-services" --> "such as web services"

"rate limited web api" is not clear. Is rate limiting web API something that each user will understand?

File: nservicebus/upgrades/5to6.md [hunk 3]
```diff
@@ -183,4 +183,8 @@ NServiceBus version 6 is no longer scanning nested directories for assemblies. Y
 
 ### Include moved to Exclude
 
-In version 6 the API has been changes to a "Exclude a list" approach. See [Assemblies to scan](/nservicebus/hosting/assembly-scanning.md#assemblies-to-scan) for more information.
\ No newline at end of file
+In version 6 the API has been changes to a "Exclude a list" approach. See [Assemblies to scan](/nservicebus/hosting/assembly-scanning.md#assemblies-to-scan) for more information.
```
Comment (SeanFeldman) [inline nservicebus/upgrades/5to6.md]: "changes" --> "changed" My own typo :smiley:

**Comment by danielmarbach** [comment]
@andreasohlund Will you also address this? 

https://github.com/Particular/docs.particular.net/blob/master/Snippets/Snippets_4/Transports/Throughput/Usage.cs
https://github.com/Particular/docs.particular.net/blob/master/Snippets/Snippets_5/Transports/Throughput/Usage.cs

I was looking for impact when we made `UnicastBusInternal`. I think those snippets should be addressed in this PR

**Comment by andreasohlund** [comment]
Will do

**Comment by andreasohlund** [comment]
@SeanFeldman updated based on your comments. Please review

**Pull request merged**